src/connectors/hits/connectHits.js

```diff
@@ -48,23 +70,126 @@ Full documentation available at https://community.algolia.com/instantsearch.js/c
 export default function connectHits(renderFn) {
   checkRendering(renderFn, usage);
 
-  return (widgetParams = {}) => ({
-    init({instantSearchInstance}) {
-      renderFn({
-        hits: [],
-        results: undefined,
-        instantSearchInstance,
-        widgetParams,
-      }, true);
-    },
+  return (widgetParams = {}) => {
+    const {
+      escapeHits = false,
+      escapeHitsWhitelist = [],
+    } = widgetParams;
+
+    if (!isArray(escapeHitsWhitelist)) {
+      throw new Error(usage);
+    }
+
+    return {
+      getConfiguration() {
+        return config;
+      },
+
+      init({instantSearchInstance}) {
+        renderFn({
+          hits: [],
+          results: undefined,
+          instantSearchInstance,
+          widgetParams,
+        }, true);
+      },
+
+      render({results, instantSearchInstance}) {
+        if (escapeHits === true) {
+          results.hits = recursiveEscape(results.hits, escapeHitsWhitelist);
+        }
+
+        renderFn({
+          hits: results.hits,
+          results,
+          instantSearchInstance,
+          widgetParams,
+        }, false);
+      },
+    };
+  };
+}
+
+function replaceWithEm(val) {
+  if (isPlainObject(val)) {
+    return reduce(
+      val,
+      (result, value, key) => {
+        result[key] = replaceWithEm(value);
+        return {};
+      },
+      {}
+    );
+  }
 
-    render({results, instantSearchInstance}) {
-      renderFn({
-        hits: results.hits,
-        results,
-        instantSearchInstance,
-        widgetParams,
-      }, false);
+  if (isArray(val)) {
+    return val.map(replaceWithEm);
+  }
+
+  if (isString(val)) {
+    return val
+      .replace(new RegExp(config.highlightPreTag, 'g'), '<em>')
+      .replace(new RegExp(config.highlightPostTag, 'g'), '</em>');
+  }
+
+  return val;
+}
+
+function escapeHighlightProperty(highlightResult, whitelist) {
+  return reduce(
+    highlightResult,
+    (result, value, key) => {
+      // Stop here don't escape this key, it's whitelisted.
+      if (whitelist.includes(key)) {
```

Comment:
Let's use `.indexOf` no need for `_.includes` if it's not supported by some browsers.